File: 2015/ruby/day2/part2.rb
#!/usr/bin/ruby

begin
    File.open('input.txt', 'r') do |file|
        totalLength = 0

        file.each_line do |line|
            dimensions = line.split('x')

            width = dimensions[0].to_i
            height = dimensions[1].to_i
            length = dimensions[2].to_i

            volume = width * height * length
            perimeter1 = (width + height) * 2
            perimeter2 = (width + length) * 2
            perimeter3 = (height + length) * 2

            totalLength += volume + [perimeter1, perimeter2, perimeter3].min
        end

        puts totalLength
    end
rescue SystemCallError
    puts 'input.txt not found.'
end

File: 2016/python/day9/part2.py
#!/usr/bin/python


def decompress(string):
    length = 0

    while '(' in string:
        parts = string.split('(', 1)
        marker, string = parts[1].split(')', 1)
        size, repetitions = map(int, marker.split('x'))

        length += len(parts[0]) + decompress(string[:size]) * repetitions
        string = string[size:]

    length += len(string)
    return length


with open('input.txt') as file_handler:
    print decompress(file_handler.read().rstrip())
